src/parameterized_tendencies/radiation/RRTMGPInterface.jl

```diff
@@ -24,6 +26,7 @@ end
 struct AllSkyRadiation <: AbstractRRTMGPMode
     idealized_h2o::Bool
     idealized_clouds::Bool
+    cloud::AbstractCloud
     add_isothermal_boundary_layer::Bool
     aerosol_radiation::Bool
```

Comment:
```suggestion
struct AllSkyRadiation{AC <: AbstractCloud} <: AbstractRRTMGPMode
    idealized_h2o::Bool
    idealized_clouds::Bool
    cloud::AC
    add_isothermal_boundary_layer::Bool
    aerosol_radiation::Bool
```
the struct has to be concrete to be used on a GPU.

I also suggest you pick a name that makes it clear that this is only for radiation.